src/ConfigProvider.php

```diff
@@ -4,42 +4,244 @@
 
 namespace Laminas\Filter;
 
-use Laminas\ServiceManager\ConfigInterface;
+use Laminas\ServiceManager\Factory\InvokableFactory;
+use Laminas\ServiceManager\ServiceManager;
 
 /**
- * @psalm-import-type ServiceManagerConfigurationType from ConfigInterface
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
  */
 class ConfigProvider
 {
     /**
      * Return configuration for this component.
      *
-     * @return array<string, mixed>
+     * @return array{dependencies: ServiceManagerConfiguration, filters: ServiceManagerConfiguration}
      */
-    public function __invoke()
+    public function __invoke(): array
     {
         return [
             'dependencies' => $this->getDependencyConfig(),
+            'filters'      => $this->filters(),
```

Comment:
I explicitly kept most of the plugin manager configuration within the plugin manager.
That was to prevent more BC breaks than necessary.
As you can see, you already had to write that `CreatePluginManager` helper class to instantiate an instance of the plugin manager because of this.
I do not see a real benefit by having this config in the `ConfigProvider`. 
Only reason I have this in `laminas-cache` is due to the fact that we do have satellites there, so I'd rather move this back into the plugin manager instead.

Reply:
My reasoning for moving config out of the plugin manager was that all of those protected props for factories, aliases, etc are now deprecated - it made sense to me to deal with deprecations at the same time considering we'll be releasing a major.

My plan in future PRs is to introduce factories for filters that rely on the plugin manager such as `FilterChain`, where the plugin manager will be made a constructor dependency.

Do you think the best course of action is to revert or to continue with that plan?

Reply:
Thats why I used config constructor Argument instead. you can see that in the serializer component for example

Reply:
Thanks for the pointer - much better solution 👍